File: LeetCode/771.py
class Solution:
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        jewels = {}
        for letter in J:
            jewels[letter] = 1
            
        total = 0
        for letter in S:
            if jewels.get(letter, 0) > 0:
                total += 1
                
        return total

File: LeetCode/771.c
int numJewelsInStones(char* J, char* S) {
    int j_len = strlen(J);
    int s_len = strlen(S);
    int total = 0;
    for(int j=0; j<j_len; j++) {
        for(int s=0; s<s_len; s++) {
            if(J[j] == S[s]) total++;
        }
    }
    return total;
}